src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,232 @@
+function sayHello() {
+  return 'world';
+}
+```
+
+For this reason, web workers have historically imposed an outsized effect on the architecture of an
+application. Developers have had to create clever tooling and workarounds to make it possible to
+use web workers without giving up modern development practises. As an example, bundlers like
+Webpack embed a small module loader implementation into generated code that uses `importScripts()`
+for code loading, but wraps modules in functions to avoid variable collisions and simulate
+dependency imports and exports.
+
+
+## Enter module workers
+
+A new mode for web workers with the ergonomics and performance benefits of [JavaScript
+Modules](https://v8.dev/features/modules) is shipping in Chrome 80, called module workers. The
+Worker constructor now accepts a new `{type:"module"}` option, which changes script loading and
+execution to match `<script type="module">`.
+
+```js
+const worker = new Worker('worker.js', {
+  type: 'module'
+});
+```
+
+Since module workers are standard JavaScript Modules, they can use import and export statements. As
+with all JavaScript Modules, dependencies are only executed once in a given context (main thread,
+worker, etc.), and all future imports reference the already-executed module instance. The loading
+and execution of JavaScript Modules is also optimized by browsers because a module's dependencies
+can be loaded prior to the module being executed, entire module trees can be loaded in parallel.
+Module loading also caches parsed code, which means modules that are used on the main thread and in
+a worker only need to be parsed once.
+
+Moving to JavaScript Modules also enables the use of [dynamic
+import](https://v8.dev/features/dynamic-import) for lazy-loading code without blocking execution of
+the worker. Dynamic import is much more explicit than using `importScripts()` to load dependencies,
+since the imported module's exports are returned rather than relying on global variables.
+
+##### worker.js:
+
+```js
+import { sayHello } from './greet.js';
+addEventListener('message', e => {
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+import greetings from './data.js';
+export function sayHello() {
+  return greetings.hello;
+}
+```
+
+To ensure great performance, the old `importScripts` API is not available within module workers.
+Switching workers to use JavaScript Modules means all code is loaded in [strict
+mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode). Another
+notable change is that the value of `this` in the top-level scope of a JavaScript Module is
+`undefined`, whereas in classic workers the value is the worker's global scope. Fortunately, there
+has always been a `self` global that provides a reference to the global scope. It's available in
+all types of workers including service workers, as well as in the DOM.
+
+{% Aside %}
+Module workers also remove support for HTML-style comments.
```

Comment:
I worry the UI treatment for Aside is a little too much for this. It's more of a lark, since very few people will even realize HTML comments were supported in classic workers. I had thought of something like this but worried it was too conversational:

> Module workers also remove support for HTML-style comments. Did you know you could use HTML comments in web worker scripts?

Reply:
Or just drop it? The only reason I bring this up is because as soon as I read this I fell into a wormhole trying to figure out how HTML comments related to classic workers, and that could take someone out of the flow of digesting the actual information you want them to focus on.

Reply:
> Module workers also remove support for HTML-style comments. Did you know you could use HTML comments in web worker scripts?

A little fun and humor is welcome. I also didn't know that this was supported and was amused to find out. This little tidbit didn't personally distract me from the main flow too much. But @jeffposnick's rationale is totally on point: we avoid side information in order to help readers stay focused on the most important main message.